lib/sinatra/authorization.rb

```diff
@@ -1,9 +1,30 @@
 module Sinatra
   module Authorization
-    def authorize_with_trusted_hosts
-      helpers ::Proxy::Helpers
+    # Authorization helper for more granular authorization (such as per-route authorization)
+    #
+    # Usage:
+    #
+    #  class MyPlugin < ::Sinatra::Base
+    #    include Sinatra::Authorization::Helpers
+    #
+    #    get '/my_public_api' do
+    #      # ...
+    #    end
+    #
+    #    get 'my_secret_api' do
+    #      # Authorize using the common set of authorization methods
+    #      do_authorize_any
+    #      # ...
+    #    end
+    #  end
+    #
+    module Helpers
+      def self.included(base)
+        base.helpers ::Proxy::Helpers
+        base.helpers ::Proxy::Log
+      end
 
-      before do
+      def do_authorize_with_trusted_hosts
```

Comment:
Why do you add the `do_` prefix to it?

Reply:
to make it different than already existing methods: I don't like to have methods with the same name doing a bit different thing

Reply:
Ah, I missed that it's the helpers module vs methods on the higher level. This part of Ruby is still rather new to me.